nautobot/core/forms/fields.py

```diff
@@ -591,8 +595,16 @@ def get_bound_field(self, form, field_name):
         widget = bound_field.field.widget
         if not widget.attrs.get("data-url"):
             route = lookup.get_route_for_model(self.queryset.model, "list", api=True)
-            data_url = reverse(route)
-            widget.attrs["data-url"] = data_url
+            try:
+                data_url = reverse(route)
+                widget.attrs["data-url"] = data_url
+            except NoReverseMatch:
+                logger.error(
+                    'API route lookup "%s" failed for model %s, form field "%s" will not work properly',
+                    route,
+                    self.queryset.model.__name__,
+                    bound_field.name,
```

Comment:
Should we use `field_name` on line 567 which was passed into the method here or `bound_field.name`? In the screenshot of the PR description, `rack__n` could be a little confusing to the end user.

Although, the `route` and `self.queryset.model.__name__` contain the most important information here so nvm.

Reply:
It's a logger message, not a UI-facing message, so I'm OK with it being a bit technical and precise.